Assets/SEE/XR/RadialSelection.cs

```diff
@@ -0,0 +1,409 @@
+                }
+                else
+                {
+                    if (actionObject != null)
+                    {
+                        actionObject.SetActive(false);
+                    }
+                }
+                GlobalActionHistory.Execute((ActionStateType)ActionStateTypes.AllRootTypes.AllElements().FirstOrDefault(a => a.Name == subMenus[i].Item1));
+            }
+            else
+            {
+                numberOfRadialPart = subMenus[i].Item2.Count() + 1;
+                menuEntrys.AddRange(actions);
+                actions.Clear();
+                actions.AddRange(subMenus[i].Item2);
+                actions.Add("Back");
+                radialMenuTrigger = true;
+                subMenuPosition = radialPartCanvas.position;
+                subMenuRotation = radialPartCanvas.rotation;
+            }
+        }
+    }
+    /// <summary>
+    /// This method spawns all radial-parts for the current menu.
+    /// </summary>
+    public void SpawnRadialPart()
+    {
+        radialPartCanvas.gameObject.SetActive(true);
+        if (subMenuPosition != null && subMenuRotation != null)
+        {
+            radialPartCanvas.position = (Vector3)subMenuPosition;
+            radialPartCanvas.rotation = (Quaternion)subMenuRotation;
+            subMenuPosition = null;
+            subMenuRotation = null;
+        }
+        else
+        {
+            radialPartCanvas.position = handTransform.position + new Vector3(0, 0, 0.2f);
+            radialPartCanvas.rotation = handTransform.rotation;
+        }
+
+
+        foreach (GameObject item in spawnedParts)
+        {
+            Destroyer.Destroy(item);
+        }
+
+        spawnedParts.Clear();
+
+        for (int i = 0; i < numberOfRadialPart; i++)
+        {
+            float angle = -i * 360 / numberOfRadialPart - angleBetweenPart / 2;
+            Vector3 radialPartEulerAngle = new Vector3(0, 0, angle);
+
+            GameObject spawnRadialPart = Instantiate(radialPartPrefab, radialPartCanvas);
+            spawnRadialPart.transform.position = radialPartCanvas.position;
+            spawnRadialPart.transform.localEulerAngles = radialPartEulerAngle;
+
+            spawnRadialPart.GetComponent<Image>().fillAmount = (1 / (float)numberOfRadialPart) - (angleBetweenPart / 360);
+            if (i > (numberOfRadialPart / 2))
+            {
+                spawnRadialPart.gameObject.transform.Find("TextField").gameObject.MustGetComponent<RectTransform>().rotation =
+                    spawnRadialPart.gameObject.transform.Find("TextField").gameObject.MustGetComponent<RectTransform>().rotation * Quaternion.Euler(0, 0, 180);
+                spawnRadialPart.gameObject.transform.Find("TextField").gameObject.MustGetComponent<TextMeshProUGUI>().alignment = TextAlignmentOptions.Right;
+            }
+            spawnRadialPart.gameObject.transform.Find("TextField").gameObject.MustGetComponent<TextMeshProUGUI>().text = actions[i];
```

Comment:
The text field should be cached into a variable (maybe even the text mesh and rect transform).